jdaviz/configs/default/plugins/subset_tools/subset_tools.py

```diff
@@ -236,15 +239,18 @@ def get_regions(self, region_type=None, list_of_subset_labels=None,
             region_type = {'imviz': ['spatial'],
                            'specviz': ['spectral']}.get(self.config, ['spatial', 'spectral'])
 
-        sky_region_check = ((self.app._align_by == 'wcs' or self.config == 'cubeviz') and
-                            return_sky_region is None or return_sky_region)
-        reg_type = 'sky_region' if sky_region_check else 'region'
+        wcs_link = self.app._align_by == 'wcs' or wrt_data is not None or self.config == 'cubeviz'
+        return_sky = return_sky_region is None or return_sky_region
+
+        # If both are True or both are False
+        reg_type = 'sky_region' if wcs_link is return_sky else 'region'
```

Comment:
I removed the error message that was raised when `wcs_link is return_sky` and `wrt_data` was set because if you are in imviz and aligned by pixels but want the SkyRegion using the wcs of the data in `wrt_data`, that error message would be raised. If you think that still needs to be raised, please let me know in the tests I created where you would expect that error.

Reply:
in the case you mentioned, wouldn't `wcs_link` be `False` and `return_sky` be `True`?

Reply:
@javerbukh - any thoughts/updates from this?

Reply:
ok, is the plan to remove the `return_sky_region` kwarg in this PR or defer that for a later effort?  If it is removed, then I think that avoids all ambiguous situations automatically.  If it is not, then I think we still need errors raised when the combination of the app linking state and the passed kwargs are in conflict.

Reply:
I'll see how much work it would take to remove it. If it's a fox/black hole, I'll open up a follow-up ticket for that.

Reply:
You were right, it is much less ambiguous without the `return_sky_region` kwarg. I updated tests accordingly and have added more information to the docstring for the `get_regions` method which hopefully makes sense. I'm still unsure if `wrt_data` really encompasses the entirety of this kwarg but we can continue iterating on that.